src/OrchardCore.Modules/OrchardCore.Users/Localization/Drivers/UserLocalizationDisplayDriver.cs

```diff
@@ -0,0 +1,67 @@
+using System.Globalization;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using OrchardCore.DisplayManagement.Entities;
+using OrchardCore.DisplayManagement.Handlers;
+using OrchardCore.DisplayManagement.ModelBinding;
+using OrchardCore.DisplayManagement.Views;
+using OrchardCore.Localization;
+using OrchardCore.Users.Localization.Models;
+using OrchardCore.Users.Localization.ViewModels;
+using OrchardCore.Users.Models;
+
+namespace OrchardCore.Users.Localization.Drivers;
+
+public class UserLocalizationDisplayDriver : SectionDisplayDriver<User, UserLocalizationSettings>
+{
+    private readonly ILocalizationService _localizationService;
+    private readonly IHttpContextAccessor _httpContextAccessor;
+
+    public UserLocalizationDisplayDriver(
+        ILocalizationService localizationService,
+        IHttpContextAccessor httpContextAccessor
+        )
+    {
+        _localizationService = localizationService;
+        _httpContextAccessor = httpContextAccessor;
+    }
+
+    public override Task<IDisplayResult> EditAsync(UserLocalizationSettings section, BuildEditorContext context)
+    {
+        return Task.FromResult<IDisplayResult>(Initialize<UserLocalizationViewModel>("UserCulture_Edit", async model =>
+        {
+            var defaultCulture = await _localizationService.GetDefaultCultureAsync();
+            var supportedCultures = await _localizationService.GetSupportedCulturesAsync();
+            var currentUserCulture = _httpContextAccessor.HttpContext.User.FindFirstValue("culture");
```

Comment:
`UserLocalizationRequestCultureProvider` also has a similar line. DRY the HCA lookup into an extension method, for example, so it's harder to make mistakes.

Reply:
I just introduced a constant to avoid the claim type typo

```csharp
internal const string CultureClaimType = "culture";
```

Reply:
I don't see the value in copying the rest of this around.

Reply:
I don't think creating an extension for a line of code is make sense

Reply:
Why would it make less sense than using the constant for a part of it?